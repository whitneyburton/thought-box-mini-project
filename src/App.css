* {
  margin: 0;
  padding: 0;
  background-color: whitesmoke;
}

.App {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  padding-top: 25px;
  /* background-image: url("./thought-bubble.jpg");
  background-position: center;
  background-repeat: no-repeat;
  background-size: 620px 228px; */

}

.thoughtbox-title {
  color: #1b446c;
  letter-spacing: 6px;
  font-size: 50px;
  padding-bottom: 15px;

}

.buttons {

}

.buttons:hover {
  cursor: pointer;
}