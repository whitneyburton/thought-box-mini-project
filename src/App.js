import React, { Component } from 'react';
import './App.css';
import CreateThought from './createThought/createThought';
import { ThoughtList } from './thoughtList/thoughtList';

class App extends Component {
  constructor() {
    super();
    this.state = {
      thoughts: []
    };
  }

  createThought = (thought) => {
    const { thoughts } = this.state;
    const newThought = { ...thought, id: Date.now() };
    this.setState({ thoughts: [...thoughts, newThought] });
  }

  deleteThought = (event) => {
    const id = parseInt(event.target.id);
    const thoughts = [...this.state.thoughts];
    let filteredThoughts = thoughts.filter(thought => thought.id !== id);
    this.setState({ thoughts: filteredThoughts });
  }

  render() {
    return (
      <div className="App">
        <div className="header">
          <h2>ThoughtBox</h2>
        </div>
        <CreateThought
          createThought={this.createThought} />
        <div>
          <ThoughtList
            thoughtList={this.state.thoughts}
            deleteThought={this.deleteThought}
            editThought={this.editThought}
          />
        </div>
      </div>
    );
  }
}

export default App;
