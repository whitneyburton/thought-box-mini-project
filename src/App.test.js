import React from 'react';
import { shallow } from 'enzyme';
import App from './App';


it('matches the snapshot', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});

it('renders a thoughtList with the correct props', () => {
  const mockThoughtList = { thoughts: [{}, {}]}
  const wrapper = shallow(<App thoughtList={mockThoughtList}/>)
  expect(wrapper.instance().props.thoughtList).toBe(mockThoughtList)
});

it('initial state is an empty array', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.state('thoughts')).toHaveLength(0);
});

it('createThought adds a thought to state', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.state('thoughts')).toHaveLength(0);
  wrapper.instance().createThought({ title: 'abc', body: 'xyz', id: 1279});
  expect(wrapper.state('thoughts')).toHaveLength(1);
});

it('deletes the correct thought from state when deleteThought is called', () => {
  const thoughts = [{ id: 0 }, { id: 2 }, { id: 3 }];
  const wrapper = shallow(<App />);
  wrapper.setState({ thoughts });
  expect(wrapper.state('thoughts')).toHaveLength(3);
  wrapper.instance().deleteThought({ target: { id: 0 } });
  expect(wrapper.state('thoughts')).toHaveLength(2);
});