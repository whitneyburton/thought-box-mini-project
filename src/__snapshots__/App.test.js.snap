// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="header"
        >
          <h2>
            ThoughtBox
          </h2>
        </div>,
        <CreateThought
          createThought={[Function]}
        />,
        <div>
          <ThoughtList
            deleteThought={[Function]}
            thoughtList={Array []}
          />
        </div>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h2>
            ThoughtBox
          </h2>,
          "className": "header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ThoughtBox",
          },
          "ref": null,
          "rendered": "ThoughtBox",
          "type": "h2",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "createThought": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ThoughtList
            deleteThought={[Function]}
            thoughtList={Array []}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "deleteThought": [Function],
            "editThought": undefined,
            "thoughtList": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header"
          >
            <h2>
              ThoughtBox
            </h2>
          </div>,
          <CreateThought
            createThought={[Function]}
          />,
          <div>
            <ThoughtList
              deleteThought={[Function]}
              thoughtList={Array []}
            />
          </div>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h2>
              ThoughtBox
            </h2>,
            "className": "header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ThoughtBox",
            },
            "ref": null,
            "rendered": "ThoughtBox",
            "type": "h2",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "createThought": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ThoughtList
              deleteThought={[Function]}
              thoughtList={Array []}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "deleteThought": [Function],
              "editThought": undefined,
              "thoughtList": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
